python/pyspark/sql/connect/functions.py

```diff
@@ -3891,6 +3894,21 @@ def udf(
 udf.__doc__ = pysparkfuncs.udf.__doc__
 
 
+def udtf(
+    cls: Optional[Type] = None,
+    *,
+    returnType: Union[StructType, str],
+    useArrow: Optional[bool] = None,
+) -> Union["UserDefinedTableFunction", functools.partial]:
```

Comment:
We typically do not use `functools.partial` as a type hint in Python. Shall we use `Callable` following the `udf` return type hints?